### PR TITLE
Add release notes for 2.10.0.0

### PR DESCRIPTION
**Pull request by msfroh**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by msfroh** [comment]
See the rendered Markdown at https://github.com/msfroh/dashboards-search-relevance/blob/release_notes_2.10/release-notes/opensearch-dashboards-search-relevance.release-notes-2.10.0.0.md

**Comment by noCharger** [comment]
> See the rendered Markdown at https://github.com/msfroh/dashboards-search-relevance/blob/release_notes_2.10/release-notes/opensearch-dashboards-search-relevance.release-notes-2.10.0.0.md

https://github.com/opensearch-project/dashboards-search-relevance/pull/278/files?short_path=f395b79#diff-f395b7969931572654f3653b144a88c55b7d67e4107257216f81e40dec16dda3 works too

**Comment by msfroh** [comment]
> > See the rendered Markdown at https://github.com/msfroh/dashboards-search-relevance/blob/release_notes_2.10/release-notes/opensearch-dashboards-search-relevance.release-notes-2.10.0.0.md
> 
> https://github.com/opensearch-project/dashboards-search-relevance/pull/278/files?short_path=f395b79#diff-f395b7969931572654f3653b144a88c55b7d67e4107257216f81e40dec16dda3 works too

Oh, nice! I didn't know about that "rich diff" feature.

File: release-notes/opensearch-dashboards-search-relevance.release-notes-2.10.0.0.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Version 2.10.0.0 Release Notes
+
+Compatible with OpenSearch 2.10.0
+
+### Enhancements
+* Changing select index error message to be more informative ([#270](https://github.com/opensearch-project/dashboards-search-relevance/pull/270)) ([#275](https://github.com/opensearch-project/dashboards-search-relevance/pull/275))
+
+### Infrastructure
+* Changing anomaly-detection Dependency Back to search-processor ([#258](https://github.com/opensearch-project/dashboards-search-relevance/pull/258))([#259](https://github.com/opensearch-project/dashboards-search-relevance/pull/259))
+* Adding a developer guide ([#268](https://github.com/opensearch-project/dashboards-search-relevance/pull/268)) ([#269](https://github.com/opensearch-project/dashboards-search-relevance/pull/269))
```
Comment (noCharger) [inline release-notes/opensearch-dashboards-search-relevance.release-notes-2.10.0.0.md:10]: should this go to Documentation? https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md#documentation

Comment (msfroh) [reply release-notes/opensearch-dashboards-search-relevance.release-notes-2.10.0.0.md:10]: Sure!

**Comment by msfroh** [comment]
Oh, we should update the developer guide to recommend using the imperative voice in commit message subject lines, as suggested in https://cbea.ms/git-commit/#imperative

**Review by noCharger — APPROVED** [review]

**Comment by noCharger** [comment]
> Oh, we should update the developer guide to recommend using the imperative voice in commit message subject lines, as suggested in https://cbea.ms/git-commit/#imperative
> 
> (And we as maintainers should enforce that convention.)

Agree, for this time we could edit the commit message on the release notes

**Pull request merged**